Assets/HoloToolkit/Utilities/Shaders/FastConfigurable/Shaders/FastConfigurable.cginc

```diff
@@ -0,0 +1,332 @@
+        #if defined(SHADOWS_SCREEN)
+        float3 lightColorShadowAttenuated = min(lightmapResult, lightAttenuation * 2);
+        #else
+        float3 lightColorShadowAttenuated = lightmapResult;
+        #endif	
+    #else
+        float3 lightColorShadowAttenuated = IN.vertexLighting;
+        #if USE_PER_PIXEL
+            //if a normal map is on, it makes sense to do most calculations per-pixel
+            #if defined(_USEBUMPMAP_ON)
+                //unpack can be expensive if normal map is dxt5
+                float3 worldNormal = UnityObjectToWorldNormal(UnpackNormal(UNITY_SAMPLE_TEX2D(_BumpMap, IN.texXYFadeZ.xy)));
+                lightColorShadowAttenuated += ShadeSH9(float4(worldNormal, 1.0));
+            #else
+                float3 worldNormal = IN.worldNormal;
+            #endif					
+        
+            #if defined(_USEDIFFUSE_ON)
+                lightColorShadowAttenuated += HoloTKLightingLambertian(worldNormal, _WorldSpaceLightPos0.xyz, _LightColor0.rgb);
+            #endif
+            #if defined(_SPECULARHIGHLIGHTS_ON)
+                float gloss = _Gloss;
+                #if defined(_USEGLOSSMAP_ON)
+                    gloss *= UNITY_SAMPLE_TEX2D(_GlossMap, IN.texXYFadeZ.xy).r;
+                #endif
+                float specular = _Specular;
+                #if defined(_USESPECULAR_ON)
+                    specular *= UNITY_SAMPLE_TEX2D(_SpecularMap, IN.texXYFadeZ.xy).r;
+                #endif
+                lightColorShadowAttenuated += HoloTKLightingBlinnPhong(worldNormal, _WorldSpaceLightPos0.xyz, _LightColor0.rgb, UnityWorldSpaceViewDir(IN.worldPos), specular, gloss, _SpecColor);
+            #endif
+            #if _SHADE4_ON
+                //handle point and directional lights
+                lightColorShadowAttenuated += Shade4PointLights(unity_4LightPosX0, unity_4LightPosY0, unity_4LightPosZ0,
+                                                                unity_LightColor[0].rgb, unity_LightColor[1].rgb, unity_LightColor[2].rgb, unity_LightColor[3].rgb,
+                                                                unity_4LightAtten0, IN.worldPos, worldNormal);
+            #endif
+            #if defined(_USERIMLIGHTING_ON)
+                lightColorShadowAttenuated += RimLight(worldNormal, IN.worldPos);
+            #endif
+        #endif
+        //could save some work here in the 0 case
+        lightColorShadowAttenuated *= lightAttenuation;
+    #endif
+    
+    color.rgb *= lightColorShadowAttenuated;
+
+    #if defined(_USEREFLECTIONS_ON)
+        float3 cubeMapColor = texCUBE(_CubeMap, IN.worldReflection);
+        color.rgb += cubeMapColor * _ReflectionScale;
+    #endif
+
+    #if defined(_USEEMISSIONMAP_ON)
+        color.rgb += UNITY_SAMPLE_TEX2D(_EmissionMap, IN.texXYFadeZ.xy);
+    #endif
+
+    #if defined(_USEEMISSIONCOLOR_ON)
+        color.rgb += _EmissionColor;
+    #endif
+
+    #if defined(_NEAR_PLANE_FADE_ON)
+        color.rgb *= IN.texXYFadeZ.z;
```

Comment:
Why don't we fade the alpha channel also? Wouldn't fading only the rgb channels result in the object blacking out instead of fading in mixed-reality capture?